src/NuGetForUnity/Editor/NugetWindow.cs

```diff
@@ -672,17 +664,17 @@ private void DrawInstalledHeader()
                     {
                         if (GUILayout.Button("Uninstall All", EditorStyles.miniButtonRight, GUILayout.Width(150)))
                         {
-                            NugetHelper.UninstallAll();
+                            NugetHelper.UninstallAll(NugetHelper.InstalledPackages.ToList());
                             NugetHelper.UpdateInstalledPackages();
                             UpdateUpdatePackages();
                         }
                     }
 
-                    if (NugetHelper.InstalledPackages.Where(p => p.IsSelected).Count() > 0)
+                    if (NugetHelper.InstalledPackages.Any(selectedPackages.Contains))
                     {
                         if (GUILayout.Button("Uninstall Selected", EditorStyles.miniButtonRight, GUILayout.Width(150)))
                         {
-                            NugetHelper.UninstallAll(true);
+                            NugetHelper.UninstallAll(NugetHelper.InstalledPackages.Where(p => selectedPackages.Contains(p)).ToList());
```

Comment:
Can you just ask `if (selectedPackages.Count > 0)` above and just do `NugetHelper.UninstallAll(selectedPackages.ToList())` here?

Reply:
@igor84 I think it is safer to `NugetHelper.UninstallAll(NugetHelper.InstalledPackages.Where(selectedPackages.Contains).ToList())` because it protects against having a package that was already uninstalled in the `selectedPackages` list. The `selectedPackages` is managed by the UI but the UI doesn't know what the `NugetHelper` dose. So I would go the safe way of only trusting the state managed by `NugetHelper`.

Reply:
```suggestion
                            NugetHelper.UninstallAll(NugetHelper.InstalledPackages.Where(selectedPackages.Contains).ToList());
```